fontir/src/ir.rs

```diff
@@ -9,22 +9,36 @@ use std::collections::{BTreeMap, HashMap};
 ///
 /// For example, upem, axis definitions, etc, as distinct from
 /// metadata that varies across design space such as ascender/descender.
-#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+#[serde(from = "StaticMetadataSerdeRepr", into = "StaticMetadataSerdeRepr")]
 pub struct StaticMetadata {
     pub axes: Vec<Axis>,
     pub glyph_order: Vec<String>,
+    pub gid_by_name: HashMap<String, u32>,
```

Comment:
fwiw fonttools calls this kind of things a glyph_map. `gid_by_name` is also fine

Reply:
if rust had a map type that remembers the insertion order we could have a single field instead of glyph_order (Vec<String>) and gid_by_name (HashMap<String, u32>); the keys of the ordered map would be the glyph order.

I see there's https://docs.rs/indexmap/latest/indexmap/map/struct.IndexMap.html 
maybe we can consider using that.

In Python 3.6+ dict keeps the insertion order, and you'll find that a lot of python font-related code relies on this inner implicitly ordered dictionary behavior.

Reply:
I dislike the python style of not being able to know if order matters. Using an explicit ordered map type selectively does address that but ... is it worth a dependency? - I'm not sure about that.

Reply:
I propose we not block the PR on this as the map is purely an accelerator (cache). If it proves annoying over time we can always update the data structure.

Reply:
it's actually very convenient in python to have its default dict type be a *hash* map type that keeps insertion order by default, so that you get constant type lookup _and_ predictable (not random) ordering when iterating. It feels in our rust code sometimes we reach out for BTreeMap just to achive that predictability at expense of O(log2(n)) lookup

Reply:
I agree re convenience, I just dislike the python style of having *all* of them retain order. Having a separate type IndexMap addresses that so maybe we should indeed do that. I really want to land this as building on unmerged PRs is painful so I am going to mvoe that to an issue.

Reply:
#54